### ISSUE TITLE
`zkevm_getBatchByNumber` has mismatched information for `accInputHash` and `timestamp`

### ISSUE THREAD

**Issue by praetoriansentry** [0xPolygon/cdk-erigon#1378]
We're cross referencing information from the RPC, Sequencer, Datastreamer, and L1 contracts for verified batches. It seems like we're often missing the `accInputHash` and potentially returning the wrong `timestamp`.

![image](https://github.com/user-attachments/assets/183caff1-acda-48dd-9cc6-0801a383695f)

In the image above, we're pulling data from a few sources. The state roots seem to align which is good, but the `accInputHash` from L1 is `0x153491ff81c722447f273fd43342407e9e5c721418c640f2102bcb8d1a7d2e67` and the RPC & Sequencer are returning `0x0000000000000000000000000000000000000000000000000000000000000000`.

Similarly, the timestamp returned by the RPC & Sequencer is `0x67214d80`, but the [timestamp](https://github.com/0xPolygonHermez/zkevm-contracts/blob/v8.1.0-rc.1-fork.13/contracts/v2/lib/LegacyZKEVMStateVariables.sol#L19-L23) returned by [getRollupSequencedBatches](https://github.com/0xPolygonHermez/zkevm-contracts/blob/v8.1.0-rc.1-fork.13/contracts/v2/PolygonRollupManager.sol)

This is happening with `hermeznetwork/cdk-erigon:v2.2.0-RC3`

**Comment by V-Staykov** [comment]
We'll need some more info: example network and batch number would be great, so we can reproduce and debug this. 

The accInputHash being zeroes should be fixed and working, if a valid contract addresses and l1 RPC url provided. It only returns zeroes if it is unable to get some data from l1 that is needed to calculate the accInputHash.

**Comment by praetoriansentry** [comment]
```bash
git clone git@github.com:0xPolygon/kurtosis-cdk.git
cd kurtosis-cdk
git checkout 210e0217771384c6177bc81cd260e1467f962bc3
kurtosis run --enclave cdk . '{"args":{"zkevm_contracts_image":"leovct/zkevm-contracts:v8.1.0-rc.1-fork.13","zkevm_prover_image":"hermeznetwork/zkevm-prover:v9.0.0-RC1-fork.13","cdk_erigon_node_image":"nulyjkdhthz/cdk-erigon:434376b78c60ff5cfdf1c6efebe813690f8d73d5"}}'
```

After letting the network get started up, give it a minute or two for some batches to get verified. You can check that the batches have climbed with this:
```bash
cast rpc --rpc-url "$(kurtosis port print cdk cdk-erigon-sequencer-001 rpc)" zkevm_verifiedBatchNumber
```

At this point we can do some checks for consistency. Assuming our last verified batch is `0x1c1`

```bash
cast rpc --rpc-url "$(kurtosis port print cdk cdk-erigon-sequencer-001 rpc)" zkevm_getBatchByNumber 0x1c | jq '.' 
cast call --json --rpc-url "$(kurtosis port print cdk el-1-geth-lighthouse rpc)" 0x2F50ef6b8e8Ee4E579B17619A92dE3E2ffbD8AD2 'getRollupSequencedBatches(uint32,uint64)(bytes32,uint64,uint64)' 1 28 | jq '.'
```
![image](https://github.com/user-attachments/assets/d2f84c89-5339-4d81-8fe6-73a530be3600)

I'm expecting that the `accInputHash` shouldn't be empty and that it would match the value from the L1 verification. Additionally, I thought the timestamp should match, but possibly not.

**Comment by zjg555543** [comment]
About `accInputHash` issue.
I added some logs with branch `upstream/v2.60.0-beta1`. It got an error while calling RPC `zkevm_getBatchByNumber`
```
failed to get acc input hash for batch 10: failed to decode calldata for tx 0x6afb234e994b2db7a15d25fea91d138bfe8fc7b09290867e909b5408188896c3: no decoder found for method signature: 165e8a8d
```
So,  I think we must to adapt decoding methods for `SequenceBatchesValidiumBanana`,`SequenceBatchesBanana`,`SequenceBatchesValidiumBanana`
https://github.com/0xPolygonHermez/cdk-erigon/blob/v2.60.0-beta1/zk/syncer/utils.go#L105

**Comment by revitteth** [comment]
Fixed by #1419  and #1421
